### PR TITLE
Ttt 24 refactor tic tac toe so that a human can play a computer opponent

### PR DESCRIPTION
**Pull request by billlanyon**
Does what it says on the tin, but lacks elegance, using several game instance variables, but no player classes.

### CODE REVIEW

File: com/bl/ttt/main.py [hunk 1]
```diff
@@ -1,21 +1,83 @@
...
+    if game.turn_player == 'X':
+        game.turn_player = 'O'
+    else:
+        game.turn_player = 'X'
+    player_id = game.turn_player
+    print(f'GP1: {game.player1}. GP2: {game.player2}. GTP: {game.turn_player}. GCP: {game.computer_player}. GTC: {game.turn_counter}.')
+    if game.computer_player is not None and game.turn_counter % 2 == 0:
+        while True:
+            computer_move = randrange(9)
+            if game.cells[computer_move] == ' ':
```
Comment (BasilBibi) [inline com/bl/ttt/main.py]: Tsk tsk you should NEVER reach into a class and look at it's state.
This should tell you that there is something wrong with your design.

Comment (billlanyon) [reply com/bl/ttt/main.py]: Oops. But I can write a method for that class called say get_turn_counter to achieve the same thing, right... or do I quit pussyfooting around and go create the player classes?

Comment (BasilBibi) [reply com/bl/ttt/main.py]: well I think maybe you should be thinking in terms of is_valid_move ?

Comment (billlanyon) [reply com/bl/ttt/main.py]: OK sir: I have attempted that.

File: com/bl/ttt/main.py [hunk 2]
```diff
@@ -1,21 +1,88 @@
...
+                print(f'Sorry, that is not a valid move for Player {move.player_id}, please try again.')
+                continue
+            break
+    if game.has_won(move.player_id):
         print(game)
-        print(f'Player {player_id} has won.')
+        print(f'Player {move.player_id} has won.')
         return game_over
     elif game.is_draw():
         print(game)
```
Comment (BasilBibi) [inline com/bl/ttt/main.py]: Is there any way to break this function down into smaller parts to make it easier to read?
Somewhere there is a metric about 5 lines of code being optimal for a function - I follow that rule losely to be honest.

File: com/bl/ttt/main.py [hunk 3]
```diff
@@ -1,21 +1,88 @@
...
+            break
+
+
+def get_first_move(game):
     player_id, cell = input('Please enter player X or O, a space, and then a cell from 0 to 8: ').split(' ')
+    game.player1 = player_id
+    if game.player1 == 'X':
+        game.player2 = 'O'
+    else:
+        game.player2 = 'X'
```
Comment (BasilBibi) [inline com/bl/ttt/main.py]: Assigning things in another class directly - also pulling things without getters

**Review by BasilBibi — APPROVED** [review]
That's good enough for now Bill well done.

**Pull request merged**